addon/data-load.js

```diff
@@ -913,12 +913,34 @@ export class Editor extends React.Component {
         ["\"", "\""],
       ]);
       const closeChar = openToCloseChar.get(e.key);
-      model.editor.setRangeText(e.key, selectionStart, selectionStart, "end");
-      model.editor.setRangeText(closeChar, selectionEnd + 1, selectionEnd + 1, "preserve");
+      // if quote (or any other start char) before and quote (or any other corresponding end char) right after it do not add quote (or the corresponding end char) but just move cursor after the it
+      if (selectionStart === selectionEnd && model.editor.value.substring(selectionStart - 1, selectionStart) == e.key && model.editor.value.substring(selectionEnd, selectionEnd + 1) == closeChar) {
+        model.editor.setRangeText("", selectionEnd + 1, selectionEnd + 1, "end");
+      } else {
+        model.editor.setRangeText(e.key, selectionStart, selectionStart, "end");
+        // add of close quote after open quote happend only if nxt character is space, break line, close parenthesis, close bracket... maybe just if next charactere is not a-z or 0-9
+        // look for char at + 1 because start char is already inserted
+        if ((e.key !== "'" && e.key !== "\"") || /[\w|\s]/.test(model.editor.value.substring(selectionEnd + 1, selectionEnd + 2))) {

```

Comment:
hmm, I think we have to split between case where we have selectionStart === selectionEnd 
and selectionStart != selectionEnd